### PR TITLE
test

### PR DESCRIPTION
**Pull request by ramya93Thigazh**
test completed but not enough time finish

### CODE REVIEW

File: pythonProject/theory questions/exam.txt [hunk 1]
```diff
@@ -0,0 +1,72 @@
+Questions:
+2-marks:
+1) What is Selenium?
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:3]: Selenium should be explained well.  1 mark.

File: pythonProject/theory questions/exam.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+selenium is a web application automation framework.
+it has more tools and libraries to automate application
+
+2) Command to install selenium.
+pip install selenium
+3) What are the locators in Selenium?
+ID,name,classname,xpath,linked text,partially linked text,css selector
+4) What is the difference between find_element_by_* and find_elements_by_* in Selenium?
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:11]: Why are we using find_element_by two times, where in the question it has been asked find_element and find_elements.? - 1mark.

File: pythonProject/theory questions/exam.txt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+find _element_by-*..... is used to find out the only single element by the locator
+ but ,find_element_by_* is used to find out the multiple element
+5) How do you manage browser windows in Selenium?
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:14]: I told you to read the question properly and answer. 
the answer for the below question is: 

driver.maximize_window()// diver.minimize_window()//driver.fullscreen_window()

No marks for this question.

File: pythonProject/theory questions/exam.txt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+driver.WindowHandles()
+driver.currentWindowHandle()
+switch_to()
+switch_to.(driver.window_handles(index))
+
+6) What are the frameworks used in selenium. Explain
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:20]: Why didn't we explain about POM.? And why we haven't shared anything technically. ? 

Giving 1 mark.

File: pythonProject/theory questions/exam.txt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+unittest-mostly used by development team
+testproject-
+robot-key driven development framework....
+maintainability,clarity,scalability,reporting these are the mager advantage in this
+
+behave(bdd)-collaboration is the best thing here..
+because of this we can collaborate with development team,QA and
+**Business partner(non technical person)...
+maintainability,clarity,simple language,gherkin keyword,collaboration are main advantage
+7) What is selenium Grid and how does it work?
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:34]: Selenium Grid is mostly used for cross browser and parallel testing. Why didnt we say this? 

0 marks.

File: pythonProject/theory questions/exam.txt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+selenium grid is used for  handle different browser in same time...
+8) How do you initialize web driver in selenium?
+By using service object
+driver=webdriver.Chrome(service=Service(ChromeWebdriverManger().install())
+9) How do you find a element in selenium?
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:39]: Why didnt we share the code snippet? 

driver.find_element(By.Xpath, 'locator'). 

0 marks.

File: pythonProject/theory questions/exam.txt [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+navigating with driver
+10) How do you handle AJAX calls in selenium.?
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:41]: Use WebDriverWait to wait for elements that are dynamically loaded after an AJAX call.

0 marks.

File: pythonProject/theory questions/exam.txt
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:1]: Total Marks: 

2 marks -- 9 /20.
10 marks -- 9/ 80.. 

total: 18/100.

File: pythonProject/theory questions/features/face.feature [hunk 8]
```diff
@@ -0,0 +1,12 @@
+Feature: Validate Facebook
+
+  Scenario Outline: validate facebook login
+    Given User opens facebook login page
+    When Enter username "<Username>" and password "<Password>" in login page
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/features/face.feature:5]: No implementation.. so No marks.

File: pythonProject/theory questions/orange.py
Comment (vickyskyrocker) [inline pythonProject/theory questions/orange.py:1]: No Marks. Question incomplete.

File: pythonProject/theory questions/ama.py
Comment (vickyskyrocker) [inline pythonProject/theory questions/ama.py:1]: 5 marks.

File: pythonProject/theory questions/exam.txt [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+DON"T KNOW SORRY
+
+10 marks:
+1) a> Use xpath and automate the below:
+    a.1> Go to amazon --> click on search box --> type iphone --> find the element in the result.
+    a.2> Go to https://the-internet.herokuapp.com/nested_frames --> identify xpath for all the frames and validate the text
+   b> Explain the git commands:
+    b.1> Git status:
```
Comment (vickyskyrocker) [inline pythonProject/theory questions/exam.txt:49]: What you will update in GIT Status. ? 

this is used to check the status of the untracked files before adding it into GIT / tracked files upon adding it into GIT.

**Review by vickyskyrocker — COMMENTED** [review]
Total marks obtained: 18/100..

**Comment by ramya93Thigazh** [comment]
merged

**Pull request merged**